src/App.tsx

```diff
@@ -26,70 +27,106 @@ type ReorderOptions = {
+      case SortType.LENGTH:
+        return itemOne.length - itemTwo.length;
+      default:
+        return 0;
+    }
+  });
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  }
+
+  sortGoods = (sortType: SortType) => () => {
+    this.setState({ sortType });
+  };
+
+  reversed = () => {
+    this.setState(prevState => ({ isReversed: !prevState.isReversed }));
+  }
+
+  reset = () => {
+    this.setState({ sortType: SortType.NONE, isReversed: false });
+  }
+
+  render() {
+    const goodies = getReorderedGoods(goodsFromServer, this.state);
+    const { isReversed, sortType } = this.state;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button is-info',
+              { 'is-light': this.state.sortType !== SortType.ALPHABET })}
+            onClick={() => this.setState({ sortType: SortType.ALPHABET })}
+          >
+            Sort alphabetically
+          </button>
+          <button
+            type="button"
+            className={classNames('button is-success',
+              { 'is-light': this.state.sortType !== SortType.LENGTH })}
+            onClick={() => this.setState({ sortType: SortType.LENGTH })}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button is-warning',
+              { 'is-light': !isReversed })}
+            onClick={this.reversed}
+          >
+            Reverse
+          </button>
+          {!isReversed && sortType === SortType.NONE
+            ? undefined
+            : (
+              <button
+                type="button"
+                className="button is-danger is-light"
+                onClick={this.reset}
+              >
+                Reset
+              </button>
+            )}
```

Comment:
why is here undefined ,use better cond rendering

Reply:
is it now how it should be?